### PR TITLE
Text field component ( input, textarea )

### PR DESCRIPTION
**Pull request by vittoriaThinkst**
## Proposed changes

Add Text Field ( Input and Text area )
Add unit tests

Screenshot 2024-03-25 at 09 28 50

### CODE REVIEW

File: frontend_vue/src/components/base/BaseTextField.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      :for="id"
+      class="mb-4 ml-4 font-semibold"
+      >{{ label }}
+      <span
+        v-if="required"
+        class="text-green-500"
+        >*</span
+      ></label
+    >
+    <component
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseTextField.vue:16]: what is a 'component'?

File: frontend_vue/src/components/base/BaseTextField.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        class="text-xs leading-4"
+      >
+        {{ helperMessage }}
+      </p>
+      <p
+        v-if="hasError"
+        id="error"
+        class="text-xs leading-4 text-red"
+      >
+        {{ errorMessage }}
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseTextField.vue:47]: maybe worth exploring an error component here we can reuse? i think this is generally the default but then folks tend to add their own spin on it.

**Review by mclmax — APPROVED** [review]

File: frontend_vue/src/components/base/BaseTextField.vue (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseTextField.vue:16]: it's a dynamic Component. You can switch between different vue components/html components via the attribute value `:is`
In this case
`<component :is="input | textarea" ...`

Textarea and input share all their properties, so in this case I just told vue to display one or the other depending on  the 'multiline' prop ( is multiline? the component should be a textarea. Is not multiline? It's just an input )

File: frontend_vue/src/components/base/BaseTextField.vue (hunk 2 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseTextField.vue:47]: I'm not sure, this is meant to be just the feedback for the input validation 🧐
We can for sure add an error component for bigger issues ( failed API calls, failed submits, etc )

**Pull request merged**